viewer/src/components/QueryExplorer.js

```diff
@@ -0,0 +1,506 @@
+              <MonacoEditor
+                height="100%"
+                language="sql"
+                theme="vs-dark"
+                value={query}
+                onChange={handleEditorChange}
+                options={{
+                  minimap: { enabled: false },
+                  scrollBeyondLastLine: false,
+                  fontSize: 14,
+                  readOnly: isLoading,
+                  automaticLayout: true,
+                  quickSuggestions: true,
+                  wordWrap: 'on',
+                  padding: { top: 16, bottom: 8 },
+                  fixedOverflowWidgets: true
+                }}
+                onMount={(editor, monaco) => {
+                  editor.setValue(query || '');
+                  
+                  // Store editor and monaco instance in refs
+                  editorRef.current = editor;
+                  monacoRef.current = monaco;
+                  
+                  // Add resize handler
+                  const resizeHandler = () => {
+                    editor.layout();
+                  };
+                  window.addEventListener('resize', resizeHandler);
+                  
+                  // Return cleanup for resize handler
+                  editor.onDidDispose(() => {
+                    window.removeEventListener('resize', resizeHandler);
+                  });
+                }}
+              />
+            </div>
+          </Panel>
+
+          <div
+            className={`h-1 bg-gray-200 hover:bg-gray-300 cursor-ns-resize flex items-center justify-center group ${
+              isDragging ? 'bg-gray-400' : ''
+            }`}
+            onMouseDown={handleMouseDown}
+          >
+            <div className="w-8 h-1 bg-gray-400 group-hover:bg-gray-500 rounded-full"></div>
+          </div>
+
+          <Panel style={{ flex: 1 - splitRatio }}>
+            <div className="flex justify-between items-center mb-4 w-full">
+              <h2 className="text-lg font-semibold">Results</h2>
+              {queryStats && (
+                <div className="text-sm text-gray-600 font-medium">
+                  {queryStats && (
+                    <>
+                      {`Last Run at ${queryStats.timestamp.toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' })} • ${queryStats.executionTime}s`}
+                      {queryStats.source && ` • Source: ${queryStats.source}`}
+                    </>
+                  )}
+                </div>
+              )}
+            </div>
+            <div className="flex-1 min-h-0 overflow-auto">
+              {results && (
+                <Table
+                  table={results}
+                  project={project}
+                  height="100%"
+                />
+              )}
+            </div>
+          </Panel>
+        </RightPanel>
+      </MainContent>
+    </Container>
+  );
+};
+
+export default QueryExplorer; 
```

Comment:
This is a pretty big component, it probably should be broken up into pieces.  The explorer is probably a good opportunity to start having global state.